common/app/uiComponent/core/JavascriptEngine.scala

```diff
@@ -0,0 +1,25 @@
+package uiComponent.core
+
+import java.io.Reader
+import javax.script._
+
+import jdk.nashorn.api.scripting.JSObject
+
+import scala.util.Try
+
+class JavascriptEngine() {
+  import JavascriptEngine._
+
+  private lazy val bindings = shared.createBindings()
```

Comment:
I remember reading this, by a developer on Nashorn: https://stackoverflow.com/a/30159424